lib/AST/ASTImporter.cpp

```diff
@@ -2201,28 +2201,29 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
         if (!ToDescribed)
           return nullptr;
         D2CXX->setDescribedClassTemplate(ToDescribed);
-        if (!DCXX->isInjectedClassName() && D->isCompleteDefinition()) {
+        if (!DCXX->isInjectedClassName()) {
           // In a record describing a template the type should be a
           // InjectedClassNameType (see Sema::CheckClassTemplate). Update the
           // previously set type to the correct value here (ToDescribed is not
           // available at record create).
           // FIXME: The previous type is cleared but not removed from
           // ASTContext's internal storage.
-          D2CXX->setTypeForDecl(nullptr);
-          Importer.getToContext().getInjectedClassNameType(D2CXX,
-              ToDescribed->getInjectedClassNameSpecialization());
-          // Find the injected class name itself and update the type too.
-          CXXRecordDecl *Injected = [&] {
+          auto Injected = [&]()->CXXRecordDecl * {
```

Comment:
Could you please list all the variables which are taken as reference in the capture list? That makes it cleaner to see which variables could be changed by the lambda.

Perhaps we should name the lambda to make its intention even cleaner. E.g. `auto FindInjectedClass = ...; auto Injected = FindInjectedClass()`

Also, formatting is a bit off here, I suppose it should be `[&]() -> CXXRecordDecl *`